File: library/src/main/java/me/tictok/hiddenwheelviewlibrary/CustomRecyclerView.java
package me.tictok.hiddenwheelviewlibrary;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;

public class CustomRecyclerView extends RecyclerView {
    public CustomRecyclerView(@NonNull Context context) {
        super(context);
    }

    public CustomRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public void attachSnapHelperWithListener(CustomSnapHelper snapHelper, OnSnapPositionChangeListener listener, SnapOnScrollListener.Behavior behavior) {
        snapHelper.attachToRecyclerView(this);
        addOnScrollListener(new SnapOnScrollListener(snapHelper, listener, behavior));
    }
}

File: library/src/main/java/me/tictok/library/CustomRecyclerView.kt
package me.tictok.library

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.util.AttributeSet

class CustomRecyclerView : RecyclerView {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle)

    fun attachSnapHelperWithListener(snapHelper: CustomSnapHelper, listener: OnSnapPositionChangeListener, behavior: SnapOnScrollListener.Behavior) {
        snapHelper.attachToRecyclerView(this)
        addOnScrollListener(SnapOnScrollListener(snapHelper, listener, behavior))
    }
}
